components/script/dom/webglrenderingcontext.rs

```diff
@@ -110,6 +110,26 @@ macro_rules! object_binding_to_js_or_null {
     };
 }
 
+macro_rules! optional_root_object_to_js_or_null {
+    ($cx: expr, $binding:expr) => {
+        {
+            rooted!(in($cx) let mut rval = NullValue());
+            if let Some(object) = $binding {
+                object.to_jsval($cx, rval.handle_mut());
+            }
+            rval.get()
+        }
+    };
+}
+
+macro_rules! bound_texture {
```

Comment:
Rather than a macro, I think a `bound_texture` method that accepts an enum argument would be more readable.

Reply:
Done